app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt

```diff
@@ -0,0 +1,29 @@
+package com.bottlerocketstudios.brarchitecture.ui
+
+import com.bottlerocketstudios.brarchitecture.ui.repository.RepositoryBrowserData
+import com.bottlerocketstudios.brarchitecture.ui.repository.RepositoryFileData
+
+@Suppress("FunctionName")
+object Routes {
+    const val Main = "main"
+    const val Home = "home"
+    const val Splash = "splash"
+    const val AuthCode = "authcode"
+    const val DevOptions = "devoptions"
+    const val Snippets = "snippets"
+    const val CreateSnippet = "snippets/create"
+    const val Profile = "profile"
+
+    fun RepositoryBrowser(data: RepositoryBrowserData)  =
+        "repository?repoName=${data.repoName}" +
+            (data.folderHash?.let { hash ->
+                "&folderHash=$hash"
+            } ?: "") +
+            (data.folderPath?.let { path ->
+                "&folderPath=$path"
+            } ?: "")
```

Comment:
@BottleRocket-Colin This could be cleaned up a little (and match the `RepositoryFile` approach below) like so:

```suggestion
    fun RepositoryBrowser(data: RepositoryBrowserData)  =
        "repository?repoName=${data.repoName}" +
            if (data.folderHash.orEmpty().isNotEmpty()) "&folderHash=${data.folderHash}" else "" +
            if (data.folderPath.orEmpty().isNotEmpty()) "&folderPath=${data.folderPath}" else ""
```